Lib/cgi.py

```diff
@@ -618,6 +618,12 @@ def read_multi(self, environ, keep_blank_values, strict_parsing):
             first_line = self.fp.readline()
             self.bytes_read += len(first_line)
 
+        # Propagate max_num_fields into the sub class appropriately
+        max_num_fields = self.max_num_fields
+        sub_max_num_fields = self.max_num_fields
+        if max_num_fields is not None:
+            sub_max_num_fields -= len(self.list)
```

Comment:
I'm not so happy for maintain two local variables.
How about keep only max_num_fields?

```py
            max_num_fields = self.max_num_fields
            if max_num_fields is not None:
                max_num_fields -= len(self.list)
```

Reply:
Oh, github removed my one comment.

```py
[snip]
part = klass(..., max_num_fields)
[snip]
if max_num_fields is not None:
    max_num_fields -= 1
    if part.list:
        max_num_fields -= len(part.list)
    if max_num_fields < 0:
        raise ValueError('Max number of fields exceeded')
```

Reply:
Good idea, that is simpler. I fixed that in a7203c8.